syntax = "proto3";

option go_package = "github.com/g13n4/go-ffmpeg-converter-microservice/video-converter";

service VideoConverter {
  rpc ConvertVideo(stream VideoToConvert) returns (stream ConvertedVideo);
}

message VideoInfo {
  string file_name = 1;
  string file_encoding = 2;
  string output_encoding = 3;
  int64 video_id = 4;
}


message VideoToConvert {
  VideoInfo video_info = 1;
  int32 chunk_index = 2;
  bool last_chuck = 3;
  bytes video_feed = 4;
}


message ConvertedVideo {
  optional bytes video_feed = 1;
  bool sucess = 2;
  optional string error = 3;
}
